grader_systems/src/stage/valgrind.rs

```diff
@@ -0,0 +1,1123 @@
+use std::{fs::File, io::Read, path::Path, time::Duration};
+
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+
+use genos::{
+    output::{Content, Output, RichTextMaker, Section, StatusUpdates, Update},
+    points::PointQuantity,
+    process::{Command, ExitStatus, ProcessExecutor, SignalType, StdinPipe},
+    stage::{StageResult, StageStatus},
+    Executor,
+};
+
+use regex::Regex;
+
+use serde::Deserialize;
+
+use tracing::debug;
+
+// default should be longer than run stage default
+const DEFAULT_TIMEOUT: Duration = Duration::from_secs(120);
+
+// exit code used to identify non-signal failures.
```

Comment:
Can you refresh my memory on how this exit code is used?

Reply:
Exit code is used to indicate a non-signal exit that had memory issues i.e. non-zero "ERROR SUMMARY". I'm not 100% sure, but it seems that ExitStatus::Signal will always be used if the process exits from a signal... the exit code is normally 128+signal on Linux, but it seems that StdExitStatus abstracts this away. I think I might need to do more testing.

We could get rid of this and rely on a non-zero exit code to determine failure (ExitStatus::Failure or ExitStatus::Signal). Right now it's used to verify that valgrind actually failed expectedly (matches the value of ExitStatus::Failure).